### PR TITLE
support named child routes

### PR DESCRIPTION
**Pull request by itajaja**
This is a little but weird, but not sure what's the best way to handle this is

### CODE REVIEW

File: src/createMatcherNamedRoutesMiddleware.js [hunk 1]
```diff
@@ -20,6 +21,10 @@ export default function createMatcherNamedRoutesMiddleware(matcher) {
     }
 
     if (children) {
+      if (!Array.isArray(children)) { // flatten route routes
+        children = Object.values(children).reduce((a, b) => a.concat(b));
```
Comment (taion) [inline src/createMatcherNamedRoutesMiddleware.js]: Why not do

```js
Object.values(children).forEach(groupChildren => {
  groupChildren.forEach(childRoute => {
    makePaths(childRoute, fullPath);
  });
});
```

Comment (itajaja) [reply src/createMatcherNamedRoutesMiddleware.js]: slightly more redundant/duplicated logic? I am fine with either

Comment (taion) [reply src/createMatcherNamedRoutesMiddleware.js]: You can factor out the callback. Concatenating all the arrays seems unnecessary.

Comment (itajaja) [reply src/createMatcherNamedRoutesMiddleware.js]: factoring out the callback doesn't increase readability imo. check out last commit

**Pull request merged**